cdap-app-fabric/src/test/java/co/cask/cdap/ConditionalWorkflowApp.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap;
+
+import co.cask.cdap.api.Predicate;
+import co.cask.cdap.api.app.AbstractApplication;
+import co.cask.cdap.api.mapreduce.AbstractMapReduce;
+import co.cask.cdap.api.mapreduce.MapReduceContext;
+import co.cask.cdap.api.workflow.AbstractWorkflow;
+import co.cask.cdap.api.workflow.WorkflowToken;
+import co.cask.cdap.internal.app.runtime.batch.WordCount;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.hadoop.mapreduce.lib.output.FileOutputFormat;
+
+import java.io.IOException;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Application to test the condition nodes in the Workflow
+ */
+public class ConditionalWorkflowApp extends AbstractApplication {
+
+  @Override
+  public void configure() {
+    setName("ConditionalWorkflowApp");
+    setDescription("Workflow app to test the Condition nodes.");
+    addWorkflow(new ConditionalWorkflow());
+    addMapReduce(new RecordVerifier());
+    addMapReduce(new WordCountMapReduce());
+  }
+
+  static final class ConditionalWorkflow extends AbstractWorkflow {
+
+    @Override
+    protected void configure() {
+      setName("ConditionalWorkflow");
+      setDescription("Workflow to test Condition nodes");
+      addMapReduce("RecordVerifier");
+      condition(new MyVerificationPredicate()).addMapReduce("ClassicWordCount").end();
```

Comment:
I think in this example, there is no else part. The complete syntax looks something like:

```java
condition(Predicate)
  .addMapReduce()
  .addMapReduce()
.otherwise()
  .addMapReduce()
.done()
```

there is no `then` because it is unnecessary and also more align with what the `fork-join` does as well as java syntax `if () { } else { }`.

The naming of the keywords `condition`, `otherwise` and `done` can be changed to whatever is more clear.